src/lib/plug.ts

```diff
@@ -1,29 +1,124 @@
-import { deserializers, serializers } from "./conversions";
+import { Accessory } from "./accessory";
 import { IPSODevice } from "./ipsoDevice";
-import { deserializeWith, ipsoKey, serializeWith } from "./ipsoObject";
+import { doNotSerialize, ipsoKey, IPSOOptions } from "./ipsoObject";
+
+export type PlugOperation = Partial<Pick<Plug,
+	"onOff"
+	>>;
 
 /* istanbul ignore next */
 export class Plug extends IPSODevice {
 
-	@ipsoKey("5805")
-	public cumulativeActivePower: number = 0.0; // <float>
+	constructor(options?: IPSOOptions, accessory?: Accessory) {
+		super(options);
+
+		// In order for the simplified API to work, the
+		// accessory reference must be a proxy
+		if (accessory != null && !accessory.isProxy) {
+			accessory = accessory.createProxy();
+		}
+		this._accessory = accessory;
+
+		// get the model number to detect features
+		if (accessory != null &&
+			accessory.deviceInfo != null &&
+			accessory.deviceInfo.modelNumber != null &&
+			accessory.deviceInfo.modelNumber.length > 0
+		) {
+			this._modelName = accessory.deviceInfo.modelNumber;
+		}
+	}
+
+	@doNotSerialize private _modelName: string | undefined;
+	@doNotSerialize private _accessory: Accessory | undefined;
 
-	@ipsoKey("5851")
-	@serializeWith(serializers.brightness)
-	@deserializeWith(deserializers.brightness)
-	public dimmer: number = 0; // <int> [0..100]
+	// @ipsoKey("5805")
```

Comment:
I'd leave all this stuff in. The keys and properties are directly taken from the app's code and might become supported in a future gateway update.
If their value is unchanged, these properties won't be sent anyways.